samples/nrf9160/modem_shell/src/ltelc/ltelc_api.c

```diff
@@ -0,0 +1,944 @@
+	if (errno) {
+		ret = -errno;
+	}
+
+	return ret;
+}
+
+#define AT_CMD_PDP_CONTEXTS_ACT_READ "AT+CGACT?"
+
+static void ltelc_api_get_pdn_states(struct pdp_context_info_array_t *pdp_info)
+{
+	char buf[16] = { 0 };
+	const char *p;
+	char at_response_str[256];
+	int ret;
+	int ctx_cnt = pdp_info->size;
+	struct pdp_context_info_t *ctx_tbl = pdp_info->array;
+
+	ret = at_cmd_write(AT_CMD_PDP_CONTEXTS_ACT_READ, at_response_str,
+			   sizeof(at_response_str), NULL);
+	if (ret) {
+		printk("Cannot get PDP contexts activation states, err: %d",
+		       ret);
+		return;
+	}
+
+	/* For each contexts: */
+	for (int i = 0; i < ctx_cnt; i++) {
+		/* Search for a string +CGACT: <cid>,<state> */
+		snprintf(buf, sizeof(buf), "+CGACT: %d,1", ctx_tbl[i].cid);
+		p = strstr(at_response_str, buf);
+		if (p) {
+			ctx_tbl[i].ctx_active = true;
+		}
+	}
+}
+
+struct pdp_context_info_t *ltelc_api_get_pdp_context_info_by_pdn_cid(int pdn_cid)
+{
+	int ret;
+	struct pdp_context_info_array_t pdp_context_info_tbl;
+	struct pdp_context_info_t *pdp_context_info = NULL;
+
+	ret = ltelc_api_pdp_contexts_read(&pdp_context_info_tbl);
+	if (ret) {
+		printk("cannot read current connection info: %d", ret);
+		return NULL;
+	}
+
+	/* Find PDP context info for the requested CID */
+	for (int i = 0; i < pdp_context_info_tbl.size; i++) {
+		if (pdp_context_info_tbl.array[i].cid == pdn_cid) {
+			pdp_context_info =
+				calloc(1, sizeof(struct pdp_context_info_t));
+			memcpy(pdp_context_info,
+			       &(pdp_context_info_tbl.array[i]),
+			       sizeof(struct pdp_context_info_t));
+			break;
+		}
+	}
+
+	if (pdp_context_info_tbl.array != NULL) {
+		free(pdp_context_info_tbl.array);
+	}
+	return pdp_context_info;
+}
+
+int ltelc_api_pdp_context_dynamic_params_get(struct pdp_context_info_t *populated_info)
+{
+	int ret = 0;
+	struct at_param_list param_list = { 0 };
+	size_t param_str_len;
+	char *next_param_str;
+	bool resp_continues = false;
+
+	char at_response_str[CONFIG_AT_CMD_RESPONSE_MAX_LEN + 1];
+	char *at_ptr = at_response_str;
+	char *tmp_ptr = at_response_str;
+	int lines = 0;
+	int iterator = 0;
+	char dns_addr_str[AT_CMD_PDP_CONTEXT_READ_IP_ADDR_STR_MAX_LEN];
+
+	char at_cmd_pdp_context_read_info_cmd_str[15];
+
+	sprintf(at_cmd_pdp_context_read_info_cmd_str,
+		AT_CMD_PDP_CONTEXT_READ_INFO, populated_info->cid);
+	ret = at_cmd_write(at_cmd_pdp_context_read_info_cmd_str,
+			   at_response_str, sizeof(at_response_str), NULL);
+	if (ret) {
+		printk("at_cmd_write returned err: %d for %s\n", ret,
+		       at_cmd_pdp_context_read_info_cmd_str);
+		return ret;
+	}
+	/* TODO: DBG logging for: printk("\n%s\n", at_response_str); */
+
+	/* Check how many rows of info do we have: */
+	while ((tmp_ptr = strstr(tmp_ptr, AT_CMD_PDP_CONTEXT_READ_RSP_DELIM)) !=
+	       NULL) {
+		++tmp_ptr;
+		++lines;
+	}
+
+	/* TODO: DBG logging for:
+	 * printf("Device contains %d lines of DNS info for CID=%d\n", lines, populated_info->cid);
+	 */
+
+	/* Parse the response: */
+	{
```

Comment:
Why is this code in a separate block?

Reply:
This style was copied from somewhere. I'm removing these blocks in this files.